docs/arch/01-deployment-modes.md

```diff
@@ -0,0 +1,479 @@
+# Deployment Modes
+
+ToolHive supports three distinct deployment modes, each optimized for different use cases and environments. This document provides a detailed explanation of how ToolHive operates in each mode.
+
+## Overview
+
+```mermaid
+graph LR
+    subgraph LocalDeployment[Local Deployment]
+        CLI[CLI Mode<br/>thv binary]
+        UI[UI Mode<br/>ToolHive Studio]
+    end
+
+    subgraph KubernetesDeployment[Kubernetes Deployment]
+        Operator[Operator Mode<br/>thv-operator]
+    end
+
+    CLI --> Docker[Docker/Podman<br/>Colima<br/>Rancher Desktop]
+    UI --> Docker
+    Operator --> K8s[Kubernetes]
+
+    Docker --> MCP1[MCP Servers]
+    K8s --> MCP2[MCP Servers]
+
+    style LocalDeployment fill:#e1f5fe
+    style KubernetesDeployment fill:#fff3e0
+```
+
+## Mode Comparison
+
+| Feature | Local CLI | Local UI | Kubernetes |
+|---------|-----------|----------|------------|
+| **Binary** | `thv` | `thv` (API server) | `thv-operator` + `thv-proxyrunner` |
+| **Container Runtime** | Docker/Podman/Colima | Docker/Podman/Rancher | Kubernetes |
```

Comment:
The Container Runtime row in this comparison table is misleading. Both Local CLI and Local UI use the same runtime detection factory and support identical container runtimes (Podman, Docker, Colima). Rancher Desktop is detected as Docker runtime type via socket path.

Consider removing this row from the table since it suggests different runtime support when they're actually the same.

Evidence: `pkg/container/factory.go` - The same factory is used for both CLI and UI modes.